applications/rag/README.md

```diff
@@ -1,192 +1,210 @@
+    * Ray may take several minutes to create the runtime environment. During this time, the job will appear to be missing (e.g. `Status message: PENDING`).
+    * Connect to the Ray dashboard to check the job status or logs:
+        - If IAP is disabled (`ray_dashboard_add_auth = false`):
+            - `kubectl port-forward -n rag service/ray-cluster-kuberay-head-svc 8265:8265`
+            - Go to `localhost:8265` in a browser
+        - If IAP is enabled (`ray_dashboard_add_auth = true`):
+            - Fetch the domain: `terraform output ray-dashboard-managed-cert`
+            - If you used a custom domain, ensure you configured your DNS as described above. This can be skipped if using `nip.io`.
+            - Verify the domain status is `Active`:
+                - `kubectl get managedcertificates ray-dashboard-managed-cert -n rag --output jsonpath='{.status.domainStatus[0].status}'`
+                - Note: This can take up to 20 minutes to propagate.
+            - Once the domain status is Active, go to the domain in a browser and login with your Google credentials.
+            - To add additional users to your frontend application, go to [Google Cloud Platform IAP](https://console.cloud.google.com/security/iap), select the `rag/ray-cluster-kuberay-head-svc` service and add principals with the role `IAP-secured Web App User`.
+
+# Launch the frontend chat interface
+
+1. Connect to the frontend:
+    * If IAP is disabled (`frontend_add_auth = false`):
+        - Port forward to the frontend service: `kubectl port-forward service/rag-frontend -n ${NAMESPACE} 8080:8080 &`
+        - Go to `localhost:8080` in a browser
+    * If IAP is enabled (`frontend_add_auth = true`):
+        - Fetch the domain: `terraform output frontend_uri`
+        - If you used a custom domain, ensure you configured your DNS as described above. This can be skipped if using `nip.io`.
+        - Verify the domain status is `Active`:
+            - `kubectl get managedcertificates frontend-managed-cert -n rag --output jsonpath='{.status.domainStatus[0].status}'`
+            - Note: This can take up to 20 minutes to propagate.
+        - Once the domain status is Active, go to the domain in a browser and login with your Google credentials.
+        - To add additional users to your frontend application, go to [Google Cloud Platform IAP](https://console.cloud.google.com/security/iap), select the `rag/rag-frontend` service and add principals with the role `IAP-secured Web App User`.
+2. Prompt the LLM
+    * Start chatting! This will fetch context related to your prompt from the vector embeddings in the `pgvector` CloudSQL instance, augment the original prompt with the context & query the inference model (`mistral-7b`) with the augmented prompt.
+
+# Configure authenticated access via IAP (recommended)
```

Comment:
Should this be part of the Installation section? If I am following the README, I will definitely miss these instructions and I probably won't have the patience to rerun the terraform script.

Reply:
@richardsliu pointed out that this made the Installation steps really long and confusing, I tend to agree. I reference it from Installation though.